consensus/parlia/parlia.go

```diff
@@ -1538,11 +1585,30 @@ func (p *Parlia) getCurrentValidators(blockHash common.Hash, blockNum *big.Int)
 		return nil, nil, err
 	}
 
-	voteAddrmap := make(map[common.Address]*types.BLSPublicKey, len(valSet))
+	voteAddrMap := make(map[common.Address]*types.BLSPublicKey, len(valSet))
+	voteAddrExist := make(map[types.BLSPublicKey]bool, len(valSet))
+
+	// after FusionBlock, there may be same validator from BC and BSC
+	// we should remove duplicated consensus address and vote address
+	// if duplicated, use the first one which has most voting power
```

Comment:
The code here does not take effect,  as you can see at the end of this function, `valSet` is directly returned without change even there is duplicated consensus addr. What is worse, it is possible that you can not get the vote addr for a consenus addr, and thus will cause `prepareValidators` failed

Reply:
Fixed

Reply:
no need to have such code here, as we can ensure it will not happen in smart contract.

Reply:
Removed